src/codegen/expression.rs

```diff
@@ -2532,48 +2361,39 @@ pub fn emit_function_call(
                 let selector = function.external_function_selector();
                 let address = function.external_function_address();
 
-                let (payload, address) = if ns.target == Target::Solana {
-                    tys.insert(0, Type::Address(false));
-                    tys.insert(1, Type::Address(false));
-                    tys.insert(2, Type::Uint(64));
-                    tys.insert(3, Type::Bytes(4));
-                    tys.insert(4, Type::Bytes(1));
-                    tys.insert(5, Type::Bytes(4));
-
-                    (
-                        Expression::AbiEncode {
-                            loc: *loc,
-                            tys,
-                            packed: vec![
-                                address,
-                                Expression::Builtin(
-                                    *loc,
-                                    vec![Type::Address(false)],
-                                    Builtin::GetAddress,
-                                    Vec::new(),
-                                ),
-                                value.clone(),
-                                Expression::NumberLiteral(*loc, Type::Bytes(4), BigInt::zero()),
-                                Expression::NumberLiteral(*loc, Type::Bytes(1), BigInt::zero()),
-                                selector,
-                            ],
-                            args,
-                        },
-                        None,
-                    )
+                let address = if ns.target == Target::Solana {
+                    let mut encoder_args: Vec<Expression> = Vec::with_capacity(6 + args.len());
```

Comment:
When a vector is allocated, it has a minimum of 8 elements, and the sized is doubled every time it runs out of capacity. 
This means that most of the time using `Vec::with_capacity()` is not really necessary. 

If we have more than 2 args, we may get one extra allocation here (more if we have more than 10 args, unlikely). So, I don't think the  `Vec::with_capacity();` followed by a bunch of pushes is really worth it vs. a single `vec![ ... ];` followed by an `extent()`. 

Anyway, up to you, I thought I would mention it.

Reply:
That is not what happens here. According to [Rust docs](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity), `with_capacity` guarantees that there is space for at least the number of elements you I requested. This way, subsequent pushes do not need reallocation. If I request `with_capacity(10)`, there will be space for ten elements and no reallocation will happen when I push ten elements into the array.

Reply:
I know how `with_capacity()` works.

What I am saying is, all this extra code does not safe much but makes the code less readable.